homeassistant/components/alexa/capabilities.py

```diff
@@ -715,4 +780,267 @@ def configuration(self):
+        if self.instance() == fan.ATTR_DIRECTION:
+            return [{"type": Catalog.LABEL_ASSET, "value": Catalog.SETTING_DIRECTION}]
+
+        return None
+
+    def mode_resources(self):
+        """Return modeResources object."""
+        mode_resources = None
+        if self.instance() == fan.ATTR_DIRECTION:
+            mode_resources = {
+                "ordered": False,
+                "resources": [
+                    {
+                        "value": f"{fan.ATTR_DIRECTION}.{fan.DIRECTION_FORWARD}",
+                        "friendly_names": [
+                            {"type": Catalog.LABEL_TEXT, "value": fan.DIRECTION_FORWARD}
+                        ],
+                    },
+                    {
+                        "value": f"{fan.ATTR_DIRECTION}.{fan.DIRECTION_REVERSE}",
+                        "friendly_names": [
+                            {"type": Catalog.LABEL_TEXT, "value": fan.DIRECTION_REVERSE}
+                        ],
+                    },
+                ],
+            }
+
+        return mode_resources
+
+    def serialize_mode_resources(self):
+        """Return ModeResources, friendlyNames serialized for an API response."""
+        mode_resources = []
+        resources = self.mode_resources()
+        ordered = resources["ordered"]
+        for resource in resources["resources"]:
+            mode_value = resource["value"]
+            friendly_names = resource["friendly_names"]
+            result = {
+                "value": mode_value,
+                "modeResources": {
+                    "friendlyNames": self.serialize_friendly_names(friendly_names)
+                },
+            }
+            mode_resources.append(result)
+
+        return {"ordered": ordered, "supportedModes": mode_resources}
+
+
+class AlexaRangeController(AlexaCapibility):
+    """Implements Alexa.RangeController.
+
+    https://developer.amazon.com/docs/device-apis/alexa-rangecontroller.html
+    """
+
+    def __init__(self, entity, instance, non_controllable=False):
+        """Initialize the entity."""
+        super().__init__(entity)
+        self.instance = lambda: instance
+        self.properties_non_controllable = lambda: non_controllable
+
+    def name(self):
+        """Return the Alexa API name of this interface."""
+        return "Alexa.RangeController"
+
+    def properties_supported(self):
+        """Return what properties this entity supports."""
+        return [{"name": "rangeValue"}]
+
+    def properties_proactively_reported(self):
+        """Return True if properties asynchronously reported."""
+        return True
+
+    def properties_retrievable(self):
+        """Return True if properties can be retrieved."""
+        return True
+
+    def get_property(self, name):
+        """Read and return a property."""
+        if name != "rangeValue":
+            raise UnsupportedProperty(name)
+
+        if self.instance() == fan.ATTR_SPEED:
```

Comment:
Why is this a function ?

Reply:
This is also not going to be an approach that is going to scale. What if 2 different domains have the same `speed` name ? We should check by domain. And in that case, maybe we can have a subclass per domain ?

Reply:
I wonder if we can even just subsclass per domain attribute. So just 1 class for speed ?

Reply:
Originally I had added `instance` as a property in the `AlexaCapibility` class defaulted to None and overridden in each subclass that needed it. But being new to the project, and a noob to Python I wasn't sure about this approach since there were no other properties in the parent class. So I just imitated what was done using in the `AlexaSceneController` with overriding the the static method with the lambda. 

Regarding the instance names, The current way it is, if another domain is added and uses the same attribute name, an additional check for the domain can be added, similar to how it's done in `get_property` for other controllers. 

I originally used `f"{fan.DOMAIN}.{fan.ATTR_SPEED}"` for instance names, but that looked unwieldy when I got to the ModeController and needed to add mode to the friendly names `f"{fan.DOMAIN}.{fan.ATTR_DIRECTION}.{fan.DIRECTION_FORWARD}"` I considered adding additional constants, or additional classes.... But ultimately just refactored for the sake of brevity for the PR figuring someone would have a better suggestion. 

I think a subclass per domain may be a good route to organize everything depending on the amount (Domains * supported_features * Controllers) are possible.

Reply:
Converted method to property, and added the entity domain for the instance identifier.